### PR TITLE
[FIX] sale_timesheet: fix activity 'upsell' when threshold is not exceed

### PR DESCRIPTION
**Pull request by ravb-odoo**
Description of the issue/feature this PR addresses:
In sale, when threshold value of product is not exceed than upsell activity is
created while creating invoice. upsell activity is only create when threshold
value exceed otherwise not.

Current behavior before PR:
When creating an invoice, an 'upsell' activity is created on the sales order and
assigned to the salesperson even if the threshold has not been exceeded.

Desired behavior after PR is merged:
Do not create a 'upsell' activity unless threshold is exceeded.

Fix:
super class is invoke before checking the upsell and create activity. so
now, we prevent super method from creating upsell activity via passing
`mail_activity_automation_skip` context.

task- 3330815

### CODE REVIEW

**Comment by xavierbol** [comment]
LGTM, can you write a test to reproduce the issue? Thanks in advance!

**Comment by ravb-odoo** [comment]
Hello @xavierbol,
I have written a test to reproduce the issue.
Thanks!!

File: addons/sale_timesheet/tests/test_upsell_warning.py [hunk 1]
```diff
@@ -86,3 +86,70 @@ def test_display_upsell_warning(self):
...
+        task = self.env['project.task'].create({
+            'name': 'Task Test',
+            'project_id': project.id,
+        })
+        task._compute_sale_line()
+
+        # 4) Timesheet in the task to satisfy the condition for the SOL to display an upsell warning
+        timesheet = self.env['account.analytic.line'].create({
+            'name': 'Test Line',
+            'unit_amount':600,
```
Comment (xavierbol) [inline addons/sale_timesheet/tests/test_upsell_warning.py:131]: 😇 
```suggestion
            'unit_amount': 600,
```

File: addons/sale_timesheet/tests/test_upsell_warning.py [hunk 2]
```diff
@@ -86,3 +86,70 @@ def test_display_upsell_warning(self):
 
         # 5) Check if the SO has an 'sale.mail_act_sale_upsell' activity.
         self.assertEqual(len(so.activity_search(['sale.mail_act_sale_upsell'])), 1, 'A upsell warning should appear in the SO.')
+
+    def test_display_upsell_warning_2(self):
```
Comment (xavierbol) [inline addons/sale_timesheet/tests/test_upsell_warning.py:90]: Your test does not fail without your fix so it means the test does not really reproduce the issue. 🤔

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I left 2 comments.

File: addons/sale_timesheet/tests/test_upsell_warning.py (hunk 2 above)
Comment (ravb-odoo) [reply addons/sale_timesheet/tests/test_upsell_warning.py]: @xavierbol ,
Please check the changes made in the test whenever you have time.
Thanks!!

File: addons/sale_timesheet/tests/test_upsell_warning.py [hunk 3]
```diff
@@ -86,3 +86,73 @@ def test_display_upsell_warning(self):
...
+            'allow_timesheets': True,
+            'allow_billable': True,
+            'partner_id': self.partner_a.id,
+            'analytic_account_id': self.analytic_account_sale.id,
+        })
+        task = self.env['project.task'].create({
+            'name': 'Task Test',
+            'project_id': project.id,
+        })
+        task._compute_sale_line()
```
Comment (xavierbol) [inline addons/sale_timesheet/tests/test_upsell_warning.py:139]: We could directly set the right SOL during the creation of the task, no? 🤔 
Moreover, I don't think we have to manually call the compute method. 🤔

File: addons/sale_timesheet/tests/test_upsell_warning.py [hunk 4]
```diff
@@ -86,3 +86,73 @@ def test_display_upsell_warning(self):
...
+
+        # 4) Timesheet in the task to satisfy the condition for the SOL to display an upsell warning
+        timesheet = self.env['account.analytic.line'].create({
+            'name': 'Test Line',
+            'unit_amount': 50,
+            'employee_id': self.employee_manager.id,
+            'project_id': project.id,
+            'task_id': task.id,
+        })
+        timesheet._compute_so_line()
```
Comment (xavierbol) [inline addons/sale_timesheet/tests/test_upsell_warning.py:149]: I don't think we have to manually call the compute. 🤔 
Alternatively, we could set the SOL during the creation of the timesheet, no? 🧐

**Review by xavierbol — COMMENTED** [review]
LGTM, thanks for your work, I just let 2 questions to be sure. 🙂

File: addons/sale_timesheet/tests/test_upsell_warning.py (hunk 4 above)
Comment (ravb-odoo) [reply addons/sale_timesheet/tests/test_upsell_warning.py]: Hello @xavierbol,
You're correct; there's no need to manually call the compute method. Instead, we can set the SOL during the task creation.
Please check the changes made in the test whenever you have time.
Thanks!!

**Review by xavierbol — APPROVED** [review]
robodoo r+

**Pull request closed without merging**